cfgmgr/buffermgrdyn.cpp

```diff
@@ -0,0 +1,1620 @@
+
+            if (table == STATE_PERIPHERAL_TABLE)
+            {
+                for (auto &fv: kfvFieldsValues(kfv))
+                {
+                    auto &field = fvField(fv);
+                    auto &value = fvValue(fv);
+                    SWSS_LOG_DEBUG("Processing table %s field:%s, value:%s", table.c_str(), field.c_str(), value.c_str());
+                    if (field == "gearbox_delay")
+                        m_gearboxDelay[key] = value;
+                }
+            }
+
+            if (table == STATE_PORT_PERIPHERAL_TABLE)
+            {
+                if (key != "global")
+                {
+                    SWSS_LOG_ERROR("Port peripheral table: only global gearbox model is supported but got %s", key.c_str());
+                    continue;
+                }
+
+                for (auto &fv: kfvFieldsValues(kfv))
+                {
+                    auto &field = fvField(fv);
+                    auto &value = fvValue(fv);
+                    SWSS_LOG_DEBUG("Processing table %s field:%s, value:%s", table.c_str(), field.c_str(), value.c_str());
+                    if (fvField(fv) == "gearbox_model")
+                        gearboxModel = fvValue(fv);
+                }
+            }
+        }
+
+        m_identifyGearboxDelay = m_gearboxDelay[gearboxModel];
+        m_supportGearbox = false;
+    }
+}
+
+void BufferMgrDynamic::initTableHandlerMap()
+{
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(STATE_BUFFER_MAXIMUM_VALUE_TABLE, &BufferMgrDynamic::handleBufferMaxParam));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_DEFAULT_LOSSLESS_BUFFER_PARAMETER, &BufferMgrDynamic::handleDefaultLossLessBufferParam));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_BUFFER_POOL_TABLE_NAME, &BufferMgrDynamic::handleBufferPoolTable));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_BUFFER_PROFILE_TABLE_NAME, &BufferMgrDynamic::handleBufferProfileTable));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_BUFFER_QUEUE_TABLE_NAME, &BufferMgrDynamic::handleBufferQueueTable));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_BUFFER_PG_TABLE_NAME, &BufferMgrDynamic::handleBufferPgTable));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_BUFFER_PORT_INGRESS_PROFILE_LIST_NAME, &BufferMgrDynamic::handleBufferPortIngressProfileListTable));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_BUFFER_PORT_EGRESS_PROFILE_LIST_NAME, &BufferMgrDynamic::handleBufferPortEgressProfileListTable));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_PORT_TABLE_NAME, &BufferMgrDynamic::handlePortTable));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_PORT_CABLE_LEN_TABLE_NAME, &BufferMgrDynamic::handleCableLenTable));
+}
+
+// APIs to handle variant kinds of keys
+
+// Transform key from CONFIG_DB format to APPL_DB format
+void BufferMgrDynamic::transformSeperator(string &name)
```

Comment:
these transform APIs could be common for both static and dynamic buffer files.

Reply:
Theoretically, yes.
To share them needs a new common class to be introduced since there is no inheriting relation between the original `BufferMgr` and `BufferMgrDyn`. It looks like an overdesign to me. So I suggest keeping as it is.